imap/index.h

```diff
@@ -96,13 +96,19 @@ struct index_init {
 struct index_map {
     modseq_t modseq;
     modseq_t told_modseq;
+    uint64_t cache_offset;
+    uint32_t user_flags[MAX_USER_FLAGS/32];
     uint32_t uid;
     uint32_t recno;
     uint32_t system_flags;
-    uint64_t cache_offset;
-    uint32_t user_flags[MAX_USER_FLAGS/32];
     unsigned int isseen:1;
     unsigned int isrecent:1;
+    /* Internal flags */
+    uint32_t flag_splitconversation:1;
```

Comment:
Why use bitfields here rather than keeping it a bit mask? Personally, I vouch for using a single uint32_t. But probably I'm not seeing the use case here?
E.g. looking at index_reload_record it looks more cumbersome to copy one flag after the other and easy to miss when adding a new flag?

Reply:
@rsto I really don't have a preference for one over the other. The intent was to make the internal flags obvious. My quick test interms of the number of instructions generated by the compiler in both cases are similar(without optimisation even)